app/src/main/java/com/aguesoftguar/medalarm/data/source/PatientsDataSource.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2016 Agüesoftgüar.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.aguesoftguar.medalarm.data.source;
+
+import android.support.annotation.NonNull;
+import com.aguesoftguar.medalarm.data.Patient;
+import java.util.List;
+
+/**
+ * Main entry point for accessing patients data.
+ */
+public interface PatientsDataSource {
+
+    /**
+     * Callbacks to load patients.
+     */
+    interface LoadPatientsCallback {
+
+        void onPatientsLoaded(List<Patient> patients);
+
+        void onDataNotAvailable(String errorDescription);
+    }
+
+    /**
+     * Callbacks to create patient.
+     */
+    interface CreatePatientsCallback {
+
+        void onPatientCreated(Patient patient);
+
+        void onAddFailed(String errorDescription);
```

Comment:
The name of these method is inconsistent because one use "create" and the other one use "add". Maybe, this is more correct:
```
void onPatientCreated(Patient patient);
void onCreatePatientFailed(String errorDescription);
```